homework/list.js

```diff
@@ -1,79 +1,108 @@
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+  ListManager.prototype = {
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
-  }
+    /**
+     * Binding the 'click' event to the '#note-list-wrapper' node and
+     * fetch the data from server then draw it.
+     * This method should be called only once to initialize the ListManager.
+     *
+     * @this {ListManager}
+     */
+    start() {
+      this.fetchList().then((function(data){
+        this.updateList(data);
+        this.drawList();
+        this.preloadFirstNote();
+      }).bind(this)).catch((function() {
 
-  function updateList(list) {
-    _listNoteContent = list;
-  }
+      }).bind(this));
+      window.addEventListener('click', (function(event) {
+        this.onNoteOpen(event);
+      }).bind(this));
+    },
 
-  function drawList() {
-    var list = _listNoteContent;
-    var ul = document.createElement('ul');
-    ul.id = 'note-title-list';
-    var buff = document.createDocumentFragment();
-    list.forEach(function(note, i) {
-      var li = document.createElement('li');
-      li.dataset.noteId = i;
-      li.classList.add('note-title');
-      li.textContent = note.title;
-      // Note: buff is captured, so we now have a
-      // little closure naturally.
-      buff.appendChild(li);
-    });
-    ul.appendChild(buff);
-    _wrapper.appendChild(ul);
-  }
+    /**
+     * Dispatch a CustomEvent 'note-open' with note detail to window for others to use.
+     *
+     * @param {CustomEvent} event - Which event.target has 'note-title' in its classList.
+     * @this {ListManager}
+     */
+    onNoteOpen(event) {
+      if (event.target.classList.contains('note-title')) {
+        var id = event.target.dataset.noteId;
+        var content = this._listNoteContent[id];
+        window.dispatchEvent(new CustomEvent('note-open',
+          { detail: content }));
+      };
+    },
 
-  function fetchList(afterFetch) {
-    var xhr = new XMLHttpRequest();
-    xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
-    xhr.responseType = 'json';
-    xhr.onreadystatechange = function(e) {
-      // Watch out: we have a mysterious unknown 'this'.
-      if (this.readyState === 4 && this.status === 200) {
-        var listData = this.response;
-        // The flow ends here.
-        afterFetch(listData);
-      } else if (this.status !== 200 ){
-        // Ignore error in this case.
+    /**
+     * Dispatch a CustomEvent 'note-open' with the first note.
+     *
+     * @this {ListManager}
+     */
+    preloadFirstNote() {
+      if (this._listNoteContent.length !== 0) {
```

Comment:
If a function body is wrapped by a conditional statement only with one branch, the `if...else` should be moved into the caller, rather than the callee.